src/views/dialogs/ComplianceConfigDialog.tsx

```diff
@@ -0,0 +1,124 @@
+import { useMemo } from 'react';
+
+import { shallowEqual, useDispatch, useSelector } from 'react-redux';
+import styled, { AnyStyledComponent } from 'styled-components';
+
+import type { Compliance } from '@/constants/abacus';
+import { ComplianceStatus } from '@/constants/abacus';
+import { ButtonAction } from '@/constants/buttons';
+import { BLOCKED_COUNTRIES, CountryCodes, OFAC_SANCTIONED_COUNTRIES } from '@/constants/geo';
+
+import { useAccounts, useDydxClient } from '@/hooks';
+
+import { Button } from '@/components/Button';
+import { ComboboxDialogMenu } from '@/components/ComboboxDialogMenu';
+import { Switch } from '@/components/Switch';
+
+import { setCompliance } from '@/state/account';
+import { getComplianceStatus, getGeo } from '@/state/accountSelectors';
+
+const complianceStatusOptions = [
+  { status: ComplianceStatus.COMPLIANT, label: 'Compliant' },
+  { status: ComplianceStatus.BLOCKED, label: 'Blocked' },
+  { status: ComplianceStatus.CLOSE_ONLY, label: 'Close Only' },
+  { status: ComplianceStatus.FIRST_STRIKE, label: 'First Strike' },
+];
+
+export const usePreferenceMenu = () => {
+  const dispatch = useDispatch();
+
+  const complianceStatus = useSelector(getComplianceStatus, shallowEqual);
+  const geo = useSelector(getGeo, shallowEqual);
+  const geoRestricted = Boolean(
+    geo && [...BLOCKED_COUNTRIES, ...OFAC_SANCTIONED_COUNTRIES].includes(geo as CountryCodes)
+  );
+
+  const notificationSection = useMemo(
+    () => ({
+      group: 'status',
+      groupLabel: 'Simulate Compliance Status',
+      items: complianceStatusOptions.map(({ status, label }) => ({
+        value: status,
+        label: label,
+        onSelect: () => dispatch(setCompliance({ geo, status } as Compliance)),
+        slotAfter: (
+          <Switch
+            name="CompliaceStatus"
+            checked={complianceStatus === status}
+            onCheckedChange={() => null} // Assuming the onChange logic is to be defined or is unnecessary
+          />
+        ),
+      })),
+    }),
+    [complianceStatus, setCompliance]
+  );
+
+  const otherSection = useMemo(
+    () => ({
+      group: 'Geo',
+      items: [
+        {
+          value: 'RestrictGeo',
```

Comment:
should this be a different value?

Reply:
This value is not being used I think. Just relying purely on the onselect to dispatch accordingly.